src/model/bpmn/internal/shape/utils.ts

```diff
@@ -28,76 +28,76 @@ import { ShapeBpmnElementKind, ShapeBpmnEventDefinitionKind } from './kinds';
  * @category BPMN
  * @experimental
  */
-export class ShapeUtil {
-  static isEvent(kind: ShapeBpmnElementKind | string): boolean {
+export const ShapeUtil = {
```

Comment:
**issue:** There is a breaking change here which is not mention explicitly.
For instance, this will break https://github.com/process-analytics/bv-experimental-add-ons/blob/v0.5.0/packages/addons/src/bpmn-elements.ts#L75-L78

Let's talk about it. We already did such a change in the past (StyleDefault, BpmnIdentifiers) but as ShapeUtil is part of the public API, we must check that in a few versions, we won't add a feature that will require to move back to the `class` construct.

Reply:
![](https://github.com/process-analytics/bpmn-visualization-js/assets/4921914/efe2c13f-9533-4e08-9489-840a4c6bc2e1)

Reply:
I am fully aware of this rule: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-static-only-class.md. This is why I switched `BpmnStyleIdentifier` from class to value object in #2716 (on 2023-05-30).

My comment does not concern this point.
It is about the breaking change that is currently being silently introduced.

> I get the impression that we're mixing up the internal model and what needs to be exposed. This type shouldn't be exposed, because it's the internal model.

Let's talk about this in person because I am not sure about the "type" your are talking to
- `ShapeUtil` is a utility which has been part of the public API for a long time and for a good reason: dealing with kinds is needed everywhere to take make choices, or to configure styles (see more details below). From the point of view of the lib, it is placed in the internal model folder, but from a user point of view, it is in the bundle, exported publicly with an "experimental" label. If we have to move it internally elsewhere, let's do it, but this won't change the way is it available publicly.
- `ShapeBpmnElementKind` is used in a lot of place in the public API (for example, as part of `BpmnElementKind` which is a parameter of various methods)

### ShapeUtil usage is bpmn-visualization-examples

https://github.com/search?q=repo%3Aprocess-analytics%2Fbpmn-visualization-examples%20shapeutil&type=code

![image](https://github.com/process-analytics/bpmn-visualization-js/assets/27200110/f1c0affb-1b40-4588-94d4-03d78c790d2e)

Reply:
ℹ️ For breaking changes auto-detection, see #2536